demo-app/src/main/java/org/openmined/syft/demo/ui/FederatedCycleViewModel.kt

```diff
@@ -1,43 +1,83 @@
 
 import android.util.Log
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import org.openmined.syft.Syft
+import org.openmined.syft.demo.domain.MNISTDataRepository
+import org.openmined.syft.domain.LocalConfiguration
 import org.openmined.syft.execution.JobStatusSubscriber
+import org.openmined.syft.execution.Plan
 import org.openmined.syft.networking.datamodels.ClientConfig
 import org.openmined.syft.proto.SyftModel
 import org.openmined.syft.threading.ProcessSchedulers
+import org.pytorch.IValue
+import java.util.concurrent.ConcurrentHashMap
 
 private const val TAG = "FederatedCycleViewModel"
-
+private const val EPOCHS = 100
 @ExperimentalUnsignedTypes
+@ExperimentalStdlibApi
 class FederatedCycleViewModel(
     baseUrl: String,
     authToken: String,
+    private val mnistDataRepository: MNISTDataRepository,
     networkSchedulers: ProcessSchedulers,
-    computeSchedulers: ProcessSchedulers
+    computeSchedulers: ProcessSchedulers,
+    private val localConfiguration: LocalConfiguration
 ) : ViewModel() {
     private val syftWorker = Syft.getInstance(
         baseUrl, authToken,
         networkSchedulers, computeSchedulers
     )
-    private val mnistJob = syftWorker.newJob("mnist", "1.0.0")
+    private val mnistJob = syftWorker.newJob("mnist", "1.0.2")
+
+    val logger
+            get() = _logger
+    private val _logger = MutableLiveData<String>()
+
 
     fun startCycle() {
         Log.d(TAG, "mnist job started")
         val jobStatusSubscriber = object : JobStatusSubscriber() {
-            override fun onReady(model: SyftModel, clientConfig: ClientConfig) {
-                //todo training code goes here
+
+            override fun onReady(
+                model: SyftModel,
+                plans: ConcurrentHashMap<String, Plan>,
+                clientConfig: ClientConfig
+            ) {
+                postLog("Model ${model.modelName} received.\nStarting training process")
+                trainingProcess(model, plans, clientConfig)
             }
 
             override fun onRejected(timeout: String) {
-                //TODO retry after timeout
+                postLog("We've been rejected $timeout")
             }
 
             override fun onError(throwable: Throwable) {
-                //todo error functionality
+                postLog("There was an error $throwable")
             }
         }
         mnistJob.start(jobStatusSubscriber)
     }
+
+    private fun postLog(message: String) {
+        _logger.postValue("${_logger.value ?: ""}\n$message")
+    }
+
+    private fun trainingProcess(
+        model: SyftModel,
+        plans: ConcurrentHashMap<String, Plan>,
+        clientConfig: ClientConfig
+    ) {
+        val destinationDir = localConfiguration.plansLocation
+
+        val plan = plans.toList().first().second
+        val loadData = mnistDataRepository.loadData()
+        repeat(EPOCHS) {
+            val output = plan.execute(model, loadData, clientConfig)
```

Comment:
You would need to call execute function on each batch/data point

Something like: 

ZippedData = zip(loadData.first,loadData.second)
zippedData.foreach{
plan.execute(model,it,clientConfig)
}